akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -138,6 +138,14 @@ private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initia
       }
       def pullFrameIn(): Unit = if (allowReadingIncomingFrames && !hasBeenPulled(frameIn)) pull(frameIn)
 
+      def tryPullSubStreams(): Unit = {
+        if (!hasBeenPulled(substreamIn) && !isClosed(substreamIn)) {
+          // While we don't support PUSH_PROMISE there's only capacity control on the client
+          if (isServer) pull(substreamIn)
+          else if (hasOutgoingCapacity) pull(substreamIn)
```

Comment:
Ah, I expected this condition inside `hasOutgoingCapacity`/`hasCapacityToCreateStreams` but checking it here is OK as well :+1: